src/event_handlers.rs

```diff
@@ -321,6 +305,67 @@ pub async fn handle_peer_name_event(peer_name: PeerName) {
     );
 }
 
+/// Handle request-response events for direct messaging
+pub async fn handle_request_response_event(
+    event: request_response::Event<DirectMessageRequest, DirectMessageResponse>,
+    swarm: &mut Swarm<StoryBehaviour>,
+    local_peer_name: &Option<String>,
+    ui_logger: &UILogger,
+) {
+    match event {
+        request_response::Event::Message { peer, message, .. } => {
+            match message {
+                request_response::Message::Request { request, channel, .. } => {
+                    // Handle incoming direct message request
+                    if let Some(local_name) = local_peer_name {
```

Comment:
The incoming message handling only checks if the to_name matches the local peer name, but doesn't validate the sender's identity.